pkg/restapi/didmethod/operation/operations.go

```diff
@@ -134,31 +142,68 @@ func (o *Operation) registerDIDHandler(rw http.ResponseWriter, req *http.Request
 		}
 
 		if v.Recovery {
-			opts = append(opts, create.WithRecoveryPublicKey(k))
+			if err := o.keyRetriever.keys.Set(didDoc.ID+"_RecoveryPublicKey", k); err != nil {
```

Comment:
I expected this to be more like did metadata - struct/map or maybe even interface that holds necessary information required for managing did.

Reply:
Also you cannot manage multiple recovery keys this way. Sidetree group recommendation is to hold onto previous recovery key until you confirm that your transaction is successful.

Reply:
Of course, you will still need key retriever to retrieve keys associated with did metadata.

Reply:
removed key retriever for create DID